db/event_handler_test.go

```diff
@@ -25,10 +27,64 @@ func TestSanitizedUrl(t *testing.T) {
+	event := &DocumentChangeEvent{DocID: docId, DocBytes: bodyBytes, OldDoc: oldBodyJSON, Channels: channels}
+
+	// Boolean return type handling of CallValidateFunction; bool true value.
+	source := `function(doc) { if (doc.key1 == "value1") { return true; } else { return false; } }`
+	filterFunc := NewJSEventFunction(source)
+	result, err := filterFunc.CallValidateFunction(event)
+	assert.True(t, result, "It should return true since doc.key1 is value1")
+	assert.NoError(t, err, "It should return boolean result")
+
+	// Boolean return type handling of CallValidateFunction; bool false value.
+	source = `function(doc) { if (doc.key1 == "value2") { return true; } else { return false; } }`
+	filterFunc = NewJSEventFunction(source)
+	result, err = filterFunc.CallValidateFunction(event)
+	assert.False(t, result, "It should return false since doc.key1 is not value2")
+	assert.NoError(t, err, "It should return boolean result")
+
+	// Parsable boolean string return type handling of CallValidateFunction.
+	source = `function(doc) { if (doc.key1 == "value1") { return "true"; } else { return "false"; } }`
+	filterFunc = NewJSEventFunction(source)
+	result, err = filterFunc.CallValidateFunction(event)
+	assert.True(t, result, "It should return true since doc.key1 is value1")
+	assert.NoError(t, err, "It should return parsable boolean result")
+
+	// Non parsable boolean string return type handling of CallValidateFunction.
+	source = `function(doc) { if (doc.key1 == "value1") { return "TrUe"; } else { return "false"; } }`
+	filterFunc = NewJSEventFunction(source)
+	result, err = filterFunc.CallValidateFunction(event)
+	assert.False(t, result, "It should return false since 'TrUe' is non parsable boolean string")
+	assert.Error(t, err, "It should return parsable throw ParseBool error")
+	assert.Contains(t, err.Error(), `strconv.ParseBool: parsing "TrUe": invalid syntax`)
+
+	// Not boolean and not parsable boolean string return type handling of CallValidateFunction.
+	source = `function(doc) { if (doc.key1 == "Pi") { return 3.14; } else { return 0.0; } }`
+	filterFunc = NewJSEventFunction(source)
+	result, err = filterFunc.CallValidateFunction(event)
+	assert.False(t, result, "It should return not boolean and not parsable boolean string value")
+	assert.Error(t, err, "It should throw Validate function returned non-boolean value error")
+	assert.Contains(t, err.Error(), "Validate function returned non-boolean value.")
+
+	// Simulate CallFunction failure by making syntax error in filter function.
+	source = `func(doc) { if (doc.key1 == "value1") { return true; } else { return false; } }`
+	filterFunc = NewJSEventFunction(source)
+	result, err = filterFunc.CallValidateFunction(event)
+	assert.False(t, result, "It should return false due to the syntax error in filter function")
+	assert.Error(t, err, "It should throw an error due to syntax error")
+	assert.Contains(t, err.Error(), "(anonymous): Line 1:12 Unexpected token { (and 4 more errors)")
```

Comment:
As discussed yesterday - it would be more future proof to reduce what we're asserting here. I think this would be sufficient for the current test data. Might need to change with the suggestion above:

```suggestion
	assert.Contains(t, err.Error(), "Unexpected token {")
```

Reply:
This has been corrected.